src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,386 @@
+    {
+        return $this->getParameter('displayName');
+    }
+
+    /**
+     * Sets the display name sent with the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setDisplayName($value)
+    {
+        return $this->setParameter('displayName', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getSignatureID()
+    {
+        return $this->getParameter('signature');
+    }
+
+    /**
+     * Sets the signature for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setSignatureID($value)
+    {
+        return $this->setParameter('signature', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getUsername()
+    {
+        return $this->getParameter('username');
+    }
+
+    /**
+     * Sets the username
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setUsername($value)
+    {
+        return $this->setParameter('username', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getPassword()
+    {
+        return $this->getParameter('password');
+    }
+
+    /**
+     * Sets the password
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setPassword($value)
+    {
+        return $this->setParameter('password', $value);
+    }
+
+    /**
+     * Overriding AbstractReuqest::sendData() so that we can make a REST call instead of a SOAP call.
+     * @param array $data
+     * @return Response
+     */
+    public function sendData($data)
+    {
+        $httpRequest  = $this->createClientRequest($data);
+        $httpResponse = $httpRequest->send();
+
+        $statusCodes = array(
+            'message' => $httpResponse->getReasonPhrase(),
+            'statusCode' => $httpResponse->getStatusCode()
+        );
+
+        try {
+            $message = $httpResponse->json();
+            $response = array_merge(
+                $statusCodes,
+                $message
+            );
+        // If you try to parse an empty response body, error will be thrown.
+        } catch (\RunTimeException $e) {
+            $response = array_merge(
+                $statusCodes,
+                array()
+            );
+        }
+
+        $this->response = new static::$RESPONSE_CLASS($this, $response);
+
+        return $this->response;
+    }
+
+    /**
+     * @param       $data
+     * @param array $headers
+     *
+     * @return \Guzzle\Http\Message\RequestInterface
+     */
+    protected function createClientRequest($data, array $headers = null)
+    {
+        $config                          = $this->httpClient->getConfig();
+        $curlOptions                     = $config->get('curl.options');
+        $curlOptions[CURLOPT_SSLVERSION] = 6;
+        // @TODO: Add path to certs and passwords. 
+        $curlOptions[CURLOPT_SSLCERT] = "YOUR_CERTS_HERE";
```

Comment:
this needs to come from a parameter. perhaps it could be set when the gateway is constructed?